server.go

```diff
@@ -1339,6 +1339,9 @@ func (s *Server) processUnaryRPC(ctx context.Context, t transport.ServerTranspor
 		if trInfo != nil {
 			trInfo.tr.LazyLog(&payload{sent: false, msg: v}, true)
 		}
+		if len(shs) == 0 && len(binlogs) == 0 {
```

Comment:
Should we think about adding a unit test for this conditional?

Reply:
@easwars

Reply:
Yes, having a unit test with a custom recv buffer pool to check that the buffer is released at the end of the RPC would be good.

Reply:
@arvindbr8 I can't figure out how to do this and how to initialize the stream for the processUnaryRPC function in order to write a unit test. Can you suggest the course of the solution?

Reply:
Maybe a more e2e style test might work better?
- Create a channel with a custom implementation of the recv buffer pool
- Perform a unary RPC
- Once the RPC is completed, ensure that the buffer is returned to your custom pool implementation

This file contains some e2e style tests which you can use as reference: https://github.com/grpc/grpc-go/blob/master/internal/idle/idle_e2e_test.go

Please let us know if you have questions. Thanks.

Reply:
@easwars Hi, there is no time to finish this task now, as there is a blockage at work and university. Can someone help with completing the task?

Reply:
@efremovmi
I've submitted a pull request and included the relevant tests. https://github.com/grpc/grpc-go/pull/6651
Please feel free to leverage it to finalize your pull request when you have a moment.
If you're unable to do so, just let me know, and I'll proceed with a pull request to close the issue.

Reply:
Hi! Did I understand correctly that we will make your change, and just close my branch? If so, then please take a look at my commit and change your code. This change is important because if an error occurs during unmarshaling, the buffer will not return to the pool

![image](https://github.com/grpc/grpc-go/assets/62056392/2dbb1689-5ec1-425f-a975-63216002381b)

Reply:
Hi!
@hueypark @easwars @arvindbr8 
Check the PR, please